### PR TITLE
fix

### PR DESCRIPTION
**Pull request by Argalep**
*  Fixed Cooldown time not working on melee creatures.
* Steal Life ability fix.
* Club weapon ability fix.

### CODE REVIEW

File: Scripts/Services/Pet Training/SpecialAbility.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public static void CheckCombatTrigger(Mobile attacker, Mobile defender, ref int
                     SpecialAbility ability = null;
 
                     SpecialAbility[] abilties = profile.EnumerateSpecialAbilities().Where(m =>
-                        (type == DamageType.Melee && m.TriggerOnDoMeleeDamage) || (type >= DamageType.Spell && m.TriggerOnDoSpellDamage) &&
```
Comment (kevin-10) [inline Scripts/Services/Pet Training/SpecialAbility.cs:44]: These brackets need to stay.

File: Scripts/Services/Pet Training/SpecialAbility.cs [hunk 2]
```diff
@@ -66,7 +66,7 @@ public static void CheckCombatTrigger(Mobile attacker, Mobile defender, ref int
                     SpecialAbility ability = null;
 
                     SpecialAbility[] abilties = profile.EnumerateSpecialAbilities().Where(m =>
-                        (type == DamageType.Melee && m.TriggerOnGotMeleeDamage) || (type >= DamageType.Spell && m.TriggerOnGotSpellDamage) &&
```
Comment (kevin-10) [inline Scripts/Services/Pet Training/SpecialAbility.cs:69]: And these.

File: Scripts/Services/Pet Training/SpecialAbility.cs [hunk 3]
```diff
@@ -1420,14 +1420,12 @@ public class StealLife : SpecialAbility
...
+        public override TimeSpan CooldownDuration => TimeSpan.FromSeconds(Utility.Random(20, 40));
 
         public override void DoEffects(BaseCreature creature, Mobile defender, ref int damage)
         {
-            defender.FixedParticles(0x374A, 1, 15, 5054, 23, 7, EffectLayer.Head);
-            defender.PlaySound(0x1F9);
-
-            var dam = AOS.Damage(defender, creature, 45, 0, 0, 0, 0, 100);
-            creature.Hits = Math.Min(creature.HitsMax, dam);
+            var dam = AOS.Damage(defender, creature, 22, 0, 0, 0, 0, 100);
```
Comment (kevin-10) [inline Scripts/Services/Pet Training/SpecialAbility.cs:1427]: according to this link: https://www.uo-cah.com/foundations/pet-abilities#lifeleech

The initial damage is 45.

**Review by kevin-10 — CHANGES_REQUESTED** [review]

File: Scripts/Services/Pet Training/SpecialAbility.cs (hunk 3 above)
Comment (Argalep) [reply Scripts/Services/Pet Training/SpecialAbility.cs:1427]: I checked in OSI. Initial damage is 22.

Comment (kevin-10) [reply Scripts/Services/Pet Training/SpecialAbility.cs:1427]: Ok, but after resists?  This is energy damage, not direct damage.

File: Scripts/Services/Pet Training/SpecialAbility.cs [hunk 4]
```diff
@@ -1454,8 +1454,8 @@ protected override void OnTick()
                 }
                 else
                 {
-                    var dam = AOS.Damage(Defender, Attacker, 5, 0, 0, 0, 0, 100);
-                    Attacker.Hits = Math.Min(Attacker.HitsMax, Attacker.Hits + dam);
+                    var dam = AOS.Damage(Defender, Attacker, 2, 0, 0, 0, 0, 100);
```
Comment (kevin-10) [inline Scripts/Services/Pet Training/SpecialAbility.cs:1457]: The link you posted a couple days ago says the initial damage is 45 and each damage tick thereafter is 5. You said you tested on EA, but were you taking into account that it was energy damage and the targets energy resists?

**Review by kevin-10 — CHANGES_REQUESTED** [review]

File: Scripts/Services/Pet Training/SpecialAbility.cs (hunk 4 above)
Comment (Argalep) [reply Scripts/Services/Pet Training/SpecialAbility.cs:1457]: I'm testing with the naked character.

You can watch the [video](https://wetransfer.com/downloads/232e72b30054d85d95ac933ead58745c20200519102145/90fd77).

**Review by kevin-10 — APPROVED** [review]

**Pull request merged**